lib/packetfu/protos/hsrp.rb

```diff
@@ -52,11 +52,10 @@ def self.can_parse?(str)
       return false unless UDPPacket.can_parse? str
       temp_packet = UDPPacket.new
       temp_packet.read(str)
-      if temp_packet.ip_ttl == 1 and [temp_packet.udp_sport,temp_packet.udp_dport] == [1985,1985] 
-        return true
-      else 
-        return false
-      end
+      return false unless temp_packet.eth_daddr == '01:00:5e:00:00:02' 
+      return false unless temp_packet.ip_daddr == '224.0.0.2' 
```

Comment:
It seems odd that we would bake this ip_daddr into this check, same with the eth_daddr.

Is the expectation that all hsrp traffic would have these values?  If so, I'd like to see some documentation/RFC/something suggesting that.  However, I get the sense this is a byproduct of you coding to your very specific use cases.

Additionally, I think it would be valuable to add some unit-tests while we're fixing this to ensure it continues to conform to expectations.

Reply:
I'm also not an HSRP expert. My work with HSRP has been based on the RFC and sample packet captures.

I've created a new PR #191 with the appropriate changes.